src/Parallel/GlobalCache.hpp

```diff
@@ -554,7 +556,25 @@ bool GlobalCache<Metavariables>::mutable_cache_item_is_ready(
                          std::vector<std::unique_ptr<Callback>>>& callbacks =
           std::get<1>(tuples::get<tag>(mutable_global_cache_));
 
-      callbacks.erase(array_component_id);
+      // It's possible that no new callbacks were registered, so make sure this
+      // array component id still has callbacks before trying to remove them.
+      if (callbacks.contains(array_component_id)) {
+        if (callbacks.at(array_component_id).empty()) {
+          callbacks.erase(array_component_id);
+        } else {
+          // If this callback was a duplicate, we'll have to search through all
+          // callbacks to determine which to remove. If it wasn't a dupliate,
+          // then it'll just be the last callback in the vector.
+          auto& vec_callbacks = callbacks.at(array_component_id);
+          auto it_to_erase = vec_callbacks.begin();
+          for (; it_to_erase != vec_callbacks.end(); it_to_erase++) {
+            if ((*it_to_erase)->is_equal_to(*clone_of_optional_callback)) {
+              vec_callbacks.erase(it_to_erase);
+              break;
+            }
+          }
```

Comment:
I think this can be done with
```cpp
vec_callbacks.erase(std::find(vec_callbacks.begin(), vec_callbacks.end(),
[&clone_of_optional_callback](const auto& t) {
  return t.is_equal_to(*clone_of_optional_callback);
});
```